core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -4142,6 +4142,14 @@ public List<SyncPointInfo> getSyncPathList() {
     return mSyncManager.getSyncPathList();
   }
 
+  @Override
+  public void forceNextSync(String path)
+      throws IOException, InvalidPathException, AccessControlException, ConnectionFailedException {
+    // TODO(jiacheng): implement this method
+    LOG.info("Invalidating cache for path {}", path);
+    mUfsSyncPathCache.invalidateCache(path);
```

Comment:
Here we have 3 options and they have subtle differences in implications. 

Option 1: Remove the entry from UfsSyncPathCache.mCache
We cannot distinguish a forced next sync from the file is just never sync-ed before. A file never being sync-ed doesn't really mean we should sync the file. The file could just be written through Alluxio. I think we still should keep the option for the user to force not to sync. 

Option 2: Distinguish a forced next sync from a never sync-ed file
Add a new special value indicating the sync should happen next time. The problem is, the UfsSyncPathCache.mCache has a size and can be evicted. If such entry is evicted then we go back to the argument whether syncInterval=-1 should be considered first.

Option 3: Just delete the file --alluxioOnly
This is actually the most direct way and makes sure the metadata sync happens. This is also how `checkConsistency` works. When the stream goes wrong we know we are deleting just a file(not a directory). The next time this file is accessed, Alluxio will check the UFS to make sure it doesn't exist. A small tickle in the brain is that we are reusing the delete semantic.